### PR TITLE
Server-Rendering new aproach for default sliders

### PR DESCRIPTION
**Pull request by joshrezende**
So i see many people who needs the component running on server side(include myself), I made some changes in the lifecycle and main logic, now on default slides we use percentage based size, this way we don't need get parent size for calculate childs width. 

the others types continue to work, running normal flow of events.

some issues about this: #273 #508

### CODE REVIEW

**Comment by lucasvmiguel** [comment]
Good Idea!

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-slick",
-  "version": "0.14.5",
+  "version": "0.14.5-beta.0",
   "description": " React port of slick carousel",
   "main": "./lib",
```
Comment (afgoulart) [inline package.json]: Add key:
"files": [ "lib/" ],

for send only available files.

**Comment by candreoliveira** [comment]
+1

@akiran how long to merge the PR?

**Comment by frankleng** [comment]
👍

**Comment by imthehell** [comment]
is the PR been merged?

**Comment by joshrezende** [comment]
@imthehell not yet, you can test from my repo, there some new improves I made on another parts, but need to push.

**Comment by imthehell** [comment]
Ok, many thanks :-) Looking forward for the merge

**Comment by bforma** [comment]
Any updates here? I'm eager on rendering react-slick universally..

**Comment by joshrezende** [comment]
@bforma sadly,  not yet... 
I'm wondering in make merge of another's PR here on Main repo and make my fork more active with the users who are making changes, what you think?

**Comment by imthehell** [comment]
I think is a good idea.

I have found another issue, the "centerMode: true" option is not working well when rendered server side. Looking on the code, the problem should be here

`ar getSlideStyle = function (spec) {
  var style = {};
  var percentUnit = spec.percentUnit;
  var sizeUnit = percentUnit ? '%' : 'px';

  if (spec.variableWidth === undefined || spec.variableWidth === false) {
    style.width = spec.slideWidth + sizeUnit;
  }

  if (spec.fade) {
    style.position = 'relative';
    style.left = -spec.index * spec.slideWidth;
    style.opacity = (spec.currentSlide === spec.index) ? 1 : 0;
    style.transition = 'opacity ' + spec.speed + 'ms ' + spec.cssEase;
    style.WebkitTransition = 'opacity ' + spec.speed + 'ms ' + spec.cssEase;
  }

  return style;
};`

The width of the slide is calculated with the % on server side and "px"on the client and the calculated widths are different. 
The following function has been added in order to calculate the slide width server-side

`getSlidePercentWidth: function(props) {
    var slideWidth = 100/(props.children.length + 2*props.slidesToShow);

    return slideWidth;
  },`

but it is not working with "centerMode: true"

Anyway, many thanks for the great work!

**Comment by joshrezende** [comment]
@iamthehell, centerMode stay in legacy functions,  I'm going to see this problem, tomorrow and post my concerns, but I think that I place him on legacy because this px units necessary for calculation.

**Comment by imthehell** [comment]
Yes it is placed on legacy functions, when I removed it from legacy I have found this problem of pixel calculation. Don't now if it is possibile to calculate the px server side..

**Comment by mkreidenweis** [comment]
What's the status on this PR? 

I'm thinking about creating a fork to at least do the initialization of `slideCount` on the server side in order to get the dots/pagers rendered on the server. 

If you're no planning to merge this, what would you think about a more minimal approach which just fixes the `slideCount` issue and maybe #508?

**Comment by joshrezende** [comment]
@mkreidenweis the PR still the same, you like I'm wondering too on create a fork, I've made some others updates on the component but using local, do you like to join forces?

We have anothers dev's seeking for server render too.

**Comment by aperkaz** [comment]
Hello, is there any news about this? We really need to have server side rendered carousel with react... 
Thanks

**Comment by aperkaz** [comment]
Hello @laveesingh ,
Why has this pull PR been closed?
Thanks

**Pull request closed without merging**